website/docs/script/spawn-direct-cross-script-calling.mdx

```diff
@@ -0,0 +1,322 @@
+---
+id: spawn-cross-script-calling
+title: "Spawn: Direct Cross-Script Calling"
+---
+
+import Tooltip from "@components/Tooltip";
+
+# Spawn: Direct Cross-Script Calling
+
+The Meepo hard fork in 2024 has introduced a range of enhancements into CKB Script development, with one major innovation being Spawn.
+
+In [CKB Script](/docs/script/intro-to-script), Spawn refers to a concept and a set of syscalls for creating new processes. Unlike the [exec syscall](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0034-vm-syscalls-2/0034-vm-syscalls-2.md#exec) introduced during [CKB’s first hard fork in 2021](/docs/history-and-hard-forks/ckb-hard-fork-history#1st-hard-fork--ckb-edition-mirana-2021), Spawn enables **direct cross-script calls**, simplifying development process while offering greater control and optimization.
+
+## Why Spawn
+
+The exec syscall, introduced into <Tooltip>CKB-VM</Tooltip> during [CKB’s first hard fork](/docs/history-and-hard-forks/ckb-hard-fork-history#1st-hard-fork--ckb-edition-mirana-2021), was designed to enable the loading and executing of a new Script within the current execution space. While powerful, exec can be inefficient, especially when preserving the current execution space is necessary. With exec, all state information of the current Script is discarded. When Script A calls Script B using exec, Script A's context is lost, as illustrated below:
+
+```
+      +----------+   Exec   +----------+
+----->+ Script A +--------->+ Script B +---->
+      +----------+          +----------+
+```
+
+We aim to preserve the context of Script A. Ideally, when Script B executes, Script A should be able to retrieve B's execution result and then continue its own execution. This process is illustrated below:
+
+```
+      +----------+
+----->+ Script A +--------------+----------->
+      +-----+----+              ^
+            |                   |
+            | Spawn             | Exit
+            v                   |
+      +----------+              |
+      | Script B +--------------+
+      +----------+
+```
```

Comment:
It seems that these ASCII text diagrams appear misaligned in the preview, despite using a monospace font (space mono).  
![Screenshot 2024-12-17 at 10 36 02 AM](https://github.com/user-attachments/assets/6a0bb120-0dba-4fac-99c6-6e1328c0cb85)

Reply:
![image](https://github.com/user-attachments/assets/dc45fdc4-1444-4053-93b8-6803bf30e299)
it seems fine on my brave browser. what is your environment?